pombola/south_africa/management/commands/south_africa_import_new_constituency_office_locations.py

```diff
@@ -0,0 +1,136 @@
+            kind__slug='constituency-office')
+
+        storage = FileSystemStorage()
+
+        storage_path = os.path.join('images', 'constituency-offices')
+        images_directory = os.path.join(
+            settings.MEDIA_ROOT,
+            storage_path
+            )
+        mkdir_p(images_directory)
+
+        with open(args[0]) as csvfile:
+            reader = csv.DictReader(csvfile)
+
+            for row in reader:
+                poi_ref = row['POI_REFERENCE']
+                party, name = (
+                    re.match(
+                        '(.*) Constituency Office (.*)', poi_ref
+                        ).groups()
+                    )
+                name = re.sub('\s+', ' ', name)
+
+                qs = constituency_offices.filter(
+                    name__regex=r'^{}.+: {}$'.format(party, name))
+
+                try:
+                    org = qs.get()
+                except Organisation.MultipleObjectsReturned:
+                    print "Skipping {} as multiple orgs returned: {}".format(
+                        poi_ref,
+                        repr(qs),
+                        )
+                    continue
+                except Organisation.DoesNotExist:
+                    # Fall back to searching for the name and the party in the
+                    # constituency office name
+                    qs = (
+                        Organisation.objects
+                        .filter(kind__slug='constituency-office')
+                        .filter(name__contains=name)
+                        .filter(name__contains=party)
+                        )
+
+                    org = qs.get()
+
+                place = (
+                    org.place_set
+                    .filter(name__contains='Approximate position of')
+                    .filter(kind__slug='constituency-office')
+                    .get()  # There should be only one.
+                    )
+
+                lon, lat = location_re.match(row['Constituency_Pin']).groups()
+
+                place.location = Point(float(lon), float(lat))
+                place.save()
+
+                # Now get the photo
+                photo_url = row.get('Constituency_Photo')
+
+                try:
+                    org.images.get(source=photo_url)
+                    print (
+                        "Skipping {} as url matches existing image."
+                        .format(org.slug)
+                        )
+                    continue
+                except Image.DoesNotExist:
+                    print "Adding new image to {}.".format(org.slug)
+
+                if photo_url:
+                    response = requests.get(photo_url)
+                    image_filename = '{}'.format(org.slug)
+
+                    desired_storage_path = os.path.join(storage_path, image_filename)
+                    storage_filename = storage.save(desired_storage_path, StringIO(response.content))
```

Comment:
Instead of using a StringIO, could you't you do:

```python
requests.get(photo_url, stream=True)
...
storage_filename = storage.save(desired_storage_path, response.raw)
```